photonjs/README.md

```diff
@@ -162,6 +166,131 @@ type Profile = {
   - Graph selection: Travesal vs operate
   - ...
 
+## Constructor
+```ts
+
+export type LogLevel = 'INFO' | 'WARN' | 'QUERY' 
+
+export type LogOption = LogLevel | {
+  level: LogLevel
+  /**
+   * @default 'stdout'
+   */
+  emit?: 'event' | 'stdout'
+}
+
+export interface PhotonOptions {
+  datasources?: Datasources
+
+  /**
+   * @default false
+   */
+  log?: boolean | LogOption[]
+
+  /**
+   * You probably don't want to use this. \`__internal\` is used by internal tooling.
+   */
+  __internal?: {
+    debug?: boolean
+    hooks?: Hooks
+    engine?: {
+      cwd?: string
+      binaryPath?: string
+    }
+  }
+}
+```
+
+## Debugging & Logging
+
+### Debugging
+
+The `debug` property includes debugging logs of the TypeScript implementation; whereas, the `log` property addresses the logs coming from the query engine.
+As the TypeScript debugging logs are mostly used by maintainers and not users, it was moved to `__internal` to not confuse users of the Photon.js library.
+
+```ts
+const photon = new Photon({
+  __internal: {
+    debug: true,
+  },
+})
+```
+
+### Logging
+
+All engine related logs can be configured with the `log` property.
+These are examples how to specify different log levels:
+
+Just providing the log levels, stdout as default
+
+```ts
+const photon = new Photon({
+  log: ['info', 'query'],
+})
+```
+
+Changing on a per log level, where the logs end up: As an event or in stdout.
+
+```ts
+const photon = new Photon({
+  log: [
+    {
+      level: 'info',
```

Comment:
This seems to be inconsistent with `export type LogLevel = 'INFO' | 'WARN' | 'QUERY'`.